Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -33,28 +33,204 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
             return
         }
 
-        let (avatarURL, isOnlineIndicatorVisible): (URL?, Bool) = {
-            // Try to get the explicit channel avatar first
-            if let avatarURL = channel.imageURL {
-                return (avatarURL, false)
+        loadAvatar(for: channel)
+    }
+    
+    // MARK: - Private
+    
+    private func loadAvatar(for channel: ChatChannel) {
+        // If the channel has an avatar set, load that avatar
+        if let channelAvatarUrl = channel.imageURL {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: channelAvatarUrl,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        // If the channel is a direct message channel, load the other user's avatar
+        // and also set the online indicator visible
+        if channel.isDirectMessageChannel {
+            let otherMember = lastActiveMembers.first
+            let isOnlineIndicatorVisible = otherMember?.isOnline ?? false
+            
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: otherMember?.imageURL,
+                placeholder: appearance.images.userAvatarPlaceholder3,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            
+            presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
+            return
+        }
+        
+        // The channel is a non-DM channel, hide the online indicator
+        presenceAvatarView.isOnlineIndicatorVisible = false
+        
+        var urls = lastActiveMembers.map(\.imageURL)
+        
+        if urls.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder3,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        } else if urls.count > 4 {
+            // We show a combination of at max 4 images combined
+            urls = Array(urls[0...3])
+        }
```

Comment:
It is probably a good idea to breakdown `loadAvatar()` into more functions and have something like:
- `loadAvatar()`:
     -`loadChannelAvatar()`
     -`loadDirectChannelAvatar()`
     -`loadMergedAvatars()`
     
So that if the customer only wants to change the logic of DM avatars, it doesn't need to override all the `loadAvatar` function 👍

Reply:
My thought process was, the customer would have just a single point of customization i.e overriding 
`open func loadAvatar(for channel: ChatChannel)`
if they want a custom implementation. Creating different methods for different cases would be a pain in terms of understanding that which is called in which conditions IMO
In the single func, they can easily make decisions based on the `channel` instance we are passing. WDYT?

Reply:
I think that pain will be for the customer needing to override and re-implement all of the `loadAvatar()` function from scratch if he is only interested in changing one little thing on for example Direct messages. We have this pattern in other parts of the code, and I think for now it is for the best. Unless we remove the early returns, then the customer can still duplicate some logic and override only for one scenario. But I think the right way to do it is to split it up into different responsibilities since it makes it easier to read, and easier to customize.

Reply:
Got your point 👍🏻  implemented separated methods accordingly